### PR TITLE
Issue2161 contributing

### LINKED ISSUE #2161: CONTRIBUTING.md and other special GitHub files
GitHub knows a couple of [special files](https://help.github.com/categories/building-a-strong-community/)
- [LICENSE.md](https://help.github.com/articles/adding-a-license-to-a-repository/) (for the license text)
- [CONTRIBUTING.md](https://help.github.com/articles/setting-guidelines-for-repository-contributors/) (for contributing guidelines)
- [ISSUE_TEMPLATE](https://help.github.com/articles/creating-an-issue-template-for-your-repository/)
- [PULL_REQUEST_TEMPLATE](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/)
- [CODE_OF_CONDUCT.md](https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/)
- .gitattributes (really from git, but github places [some things](https://github.com/github/linguist#overrides) here, too)

These files, especially the CONTRIBUTING.md, can make life easier for first-time contributors and help with an efficient workflow, describing how to properly fork a repository, create issue/topic branches, cross-reference issues, submit a pull request and so on.
Another possibly helpful feature are [Pull Request reviews](https://help.github.com/articles/about-pull-request-reviews/) in combination with [protected branches](https://help.github.com/articles/about-protected-branches/).

Do you want me to submit a PR for an inital version of a CONTRIBUTING.md file?

### PR DESCRIPTION
**Pull request by thorade**
An initial version for Contributing guidelines.
This would close #2161 for now.

### CODE REVIEW

**Comment by beutlich** [comment]
The contribution Guideline clearly is related to #2165.

**Comment by thorade** [comment]
The file contains some links already, here are some other links that might or might not be added to the file:

- Any of the guides at https://opensource.guide/
- Any of the guides at https://guides.github.com/
- Closing issues via commit message: https://help.github.com/articles/closing-issues-via-commit-messages/
- Closign issues via pull requests: https://github.com/blog/1506-closing-issues-via-pull-requests
- for reviewers: ignore whitespace and other tricks https://github.com/tiimgreen/github-cheat-sheet/blob/master/README.md#ignore-whitespace
- working with non-code files: https://help.github.com/categories/working-with-non-code-files/
- protected branches and required checks: https://help.github.com/articles/defining-the-mergeability-of-pull-requests/

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+The workflow for contributing code depends on whether you are a collaborator 
+with push rights, or an outside collaborator.
+
+The [workflow for outside collaborators](https://guides.github.com/activities/forking/) is usually as follows:  
+
+1. Fork the repository to your account by
+   [using the `Fork` button](https://help.github.com/articles/fork-a-repo/).
+2. Clone the forked repository to your computer.
+3. Create a new topic branch and give it a meaningful name, 
+   like e.g. `issue2161_contributing`.
```
Comment (dietmarw) [inline CONTRIBUTING.md]: "e.g." is always written enclosed in commas, i.e., ",e.g.,". Same for "i.e." as you see ;-)

**Comment by thorade** [comment]
@dietmarw Changed. You can also do changes yourself if you want.

**Comment by dietmarw** [comment]
I know but then how will you ever learn ;-)

**Comment by xogeny** [comment]
I don't think you need "enclosing" commas if it is parenthetical (*e.g.,* you see what I did right here?).

**Comment by dietmarw** [comment]
True :-)

**Pull request merged**